thirdparty/github.go

```diff
@@ -1333,24 +1329,34 @@ func MostRestrictiveGitHubPermission(perm1, perm2 string) string {
 // GetPullRequestMergeBase returns the merge base hash for the given PR.
 // This function will retry up to 5 times, regardless of error response (unless
 // error is the result of hitting an api limit)
-func GetPullRequestMergeBase(ctx context.Context, data GithubPatch) (string, error) {
+func GetPullRequestMergeBase(ctx context.Context, owner, repo, baseRevision, currentCommitHash string, prNum int) (string, error) {
+	mergeBase, err := GetGithubMergeBaseRevision(ctx, owner, repo, baseRevision, currentCommitHash)
+	if err == nil {
+		return mergeBase, nil
+	}
+	apiErr, ok := errors.Cause(err).(APIRequestError)
+	if !ok || apiErr.StatusCode != http.StatusNotFound {
+		return "", err
+	}
+	// If GetGithubMergeBaseRevision returns a 404 APIRequestError, fallback to the secondary way of determining a PR
+	// merge base via API.
```

Comment:
May be worth a clarification comment on situations where this is actually acceptable, like in the case of finding the merge base of PR based on a private fork that Evergreen doesn't have permission to access.